### PR TITLE
move unauthorizedHandler to AuthorizationService

### PR DESCRIPTION
**Pull request by saeideng**

### CODE REVIEW

File: src/AuthorizationService.php [hunk 1]
```diff
@@ -22,9 +22,21 @@
...
+use Cake\Core\InstanceConfigTrait;
 
 class AuthorizationService implements AuthorizationServiceInterface
 {
+    use InstanceConfigTrait;
+
+    /**
+     * Default config.
+     *
+     * @var array
```
Comment (dereuromark) [inline src/AuthorizationService.php:34]:
```suggestion
     * @var array<string, mixed>
```

File: src/Middleware/UnauthorizedHandler/CakeRedirectHandler.php [hunk 2]
```diff
@@ -25,6 +25,7 @@
  * This handler will redirect the response if one of configured exception classes is encountered.
  *
  * CakePHP Router compatible array URL syntax is supported.
+ * @deprecated use RedirectHandler instead.
```
Comment (markstory) [inline src/Middleware/UnauthorizedHandler/CakeRedirectHandler.php:28]: :clap: I agree with the service hook method developers can use `Router::url()` to generate the URL they need.

**Comment by saeideng** [comment]
do we need move all configs from  `AuthorizationMiddleware` to `AuthorizationService`
or only the `unauthorizedHandler` ?

my opinion : moving all

**Comment by markstory** [comment]
> my opinion : moving all

Would you also be moving all the logic related to these options to the service as well?

**Comment by saeideng** [comment]
> > my opinion : moving all
> 
> Would you also be moving all the logic related to these options to the service as well?


not sure , but I can try it

**Comment by ADmad** [comment]
I didn't see this PR earlier. #211 adds the `unauthorizedHandler` to both the middlewares using a trait instead.

**Pull request closed without merging**